spec/api/goals_api_spec.rb

```diff
@@ -1,6 +1,13 @@
 require 'rails_helper'
 
 describe '/api/goals' do
+  def basic_credentials(email, password)
```

Comment:
Consider reducing the arguments from `basic_credentials(email, pwd)` to `basic_credentials(user)` to avoid the repeated `user.email, user.password` calls.